modules/free/src/main/scala/doobie/free/kleisliinterpreter.scala

```diff
@@ -74,19 +75,12 @@ import doobie.free.callablestatement.{ CallableStatementIO, CallableStatementOp
 import doobie.free.resultset.{ ResultSetIO, ResultSetOp }
 
 object KleisliInterpreter {
-
-  def apply[M[_]](
-    implicit am: WeakAsync[M]
-  ): KleisliInterpreter[M] =
-    new KleisliInterpreter[M] {
-      val asyncM = am
-    }
+  def apply[M[_]: WeakAsync](logHandler: Option[LogHandlerM[M]]): KleisliInterpreter[M] =
+    new KleisliInterpreter[M](logHandler)
 }
 
 // Family of interpreters into Kleisli arrows for some monad M.
-trait KleisliInterpreter[M[_]] { outer =>
-
-  implicit val asyncM: WeakAsync[M]
+class KleisliInterpreter[M[_]](logHandler: Option[LogHandlerM[M]])(implicit val asyncM: WeakAsync[M]) { outer =>
```

Comment:
Do you think we should take just a `LogHandlerM[M]` and define a NoopLogHandlerM?
This means that in `performLogging` we won't need to pay for a `match` every time.